### PR TITLE
Add Auth namespace docs

### PR DESCRIPTION
**Pull request by HaoK**
Part of https://github.com/dotnet/aspnetcore/issues/27286

### CODE REVIEW

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureAD.UI.xml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.AzureAD.UI">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureAD.UI.xml:4]: remarks doesn't need to be removed.

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.xml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for authentication.</summary>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.xml:3]: This namespace is pretty important. It should have some more information. See some of the longer summaries at https://docs.microsoft.com/en-us/dotnet/api/?view=netcore-3.1

Also link to https://docs.microsoft.com/aspnet/core/security/authentication in the remarks:

`<remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication">Overview of ASP.NET Core authentication</see>.</remarks>`

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authorization.xml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 <Namespace Name="Microsoft.AspNetCore.Authorization">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for authorization.</summary>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authorization.xml:3]: This namespace is pretty important. It should have some more information. See some of the longer summaries at https://docs.microsoft.com/en-us/dotnet/api/?view=netcore-3.1

Also add a link in the remarks:

`<remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authorization/introduction">Introduction to authorization in ASP.NET Core</see>.</remarks>`

**Comment by HaoK** [comment]
Updated with remarks and more details in the main namespaces @JamesNK

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureAD.UI.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
+    <summary>Contains types that enable better authentication integration with Azure Active Directory.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/azure-active-directory/?view=aspnetcore-3.1">Introduction to authentication with Active Directory in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureAD.UI.xml:4]: Update to remove language from the links (so it is auto-detected from the browser) and the version number.
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/azure-active-directory/">Introduction to authentication with Active Directory in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureADB2C.UI.xml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.AzureADB2C.UI">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable better authentication integration with Azure Active Directory B2C.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/azure-active-directory/?view=aspnetcore-3.1">Introduction to authentication with Active Directory in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureADB2C.UI.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/azure-active-directory/">Introduction to authentication with Active Directory in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Certificate.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Certificate">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provides classes to enable certificate authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/certauth">Configure certificate authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Certificate.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/certauth">Configure certificate authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Cookies.xml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Cookies">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for cookie based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie">Use cookie authentication without ASP.NET Core Identity</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Cookies.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.comaspnet/core/security/authentication/cookie">Use cookie authentication without ASP.NET Core Identity</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Facebook.xml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Facebook">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for facebook OAuth based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Facebook.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Google.xml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Google">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for Google OAuth based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Google.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.JwtBearer.xml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.JwtBearer">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for JWT bearer based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication">Introduction to authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.JwtBearer.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication">Introduction to authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.MicrosoftAccount.xml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.MicrosoftAccount">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for Microsoft Account OAuth based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.MicrosoftAccount.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Negotiate.xml [hunk 12]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Negotiate">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for Negotiate (also known as Windows, Kerberos, or NTLM) based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/windowsauth">Configure Windows Authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Negotiate.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/windowsauth">Configure Windows Authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OAuth.Claims.xml [hunk 13]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.OAuth.Claims">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that make it easier to customize claim mappings with OAuth based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/additional-claims">Persist additional claims and tokens from external providers in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OAuth.Claims.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social/additional-claims">Persist additional claims and tokens from external providers in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OAuth.xml [hunk 14]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.OAuth">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for OAuth based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OAuth.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OpenIdConnect.Claims.xml [hunk 15]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.OpenIdConnect.Claims">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that make it easier to customize claim mappings with OpenIdConnect based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/additional-claims">Persist additional claims and tokens from external providers in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OpenIdConnect.Claims.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social/additional-claims">Persist additional claims and tokens from external providers in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OpenIdConnect.xml [hunk 16]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.OpenIdConnect">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for OpenIdConnect based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication">Introduction to authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.OpenIdConnect.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication">Introduction to authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Twitter.xml [hunk 17]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Twitter">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for Twitter based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Twitter.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/social">Facebook, Google, and external provider authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.WsFederation.xml [hunk 18]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.WsFederation">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for WSFederation based authentication.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication/ws-federation">Authenticate users with WS-Federation in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.WsFederation.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication/ws-federation">Authenticate users with WS-Federation in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.xml [hunk 19]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for authentication. IAuthenticationService is the main entry point which defines the 5 authentication actions: Authenticate (get any authentication data for a request), Challenge (used for unauthenticated requests), Forbid (used when an authenticated request should be dnied), SignIn (associate a ClaimsPrincipal), SignOut (remove any associated data). At a high level, authentication is based around the concept of schemes, which are mapped to specific AuthenticationHandlers which are responsible for handling each of the authentication actions.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authentication">Introduction to authentication in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authentication">Introduction to authentication in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authorization.Policy.xml [hunk 20]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authorization.Policy">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for policy based authorization.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/security/authorization/policies">Policy-based authorization in ASP.NET Core</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authorization.Policy.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/security/authorization/policies">Policy-based authorization in ASP.NET Core</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureAD.UI.xml [hunk 21]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.AzureAD.UI">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable better authentication integration with Azure Active Directory.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureAD.UI.xml:3]: Note crazy about better. Better than what? Better how. Consider:

```suggestion
    <summary>Contains types that support authentication integration with Azure Active Directory.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureADB2C.UI.xml [hunk 22]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.AzureADB2C.UI">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable better authentication integration with Azure Active Directory B2C.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.AzureADB2C.UI.xml:3]:
```suggestion
    <summary>Contains types that support authentication integration with Azure Active Directory B2C.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Certificate.xml [hunk 23]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Certificate">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provides classes to enable certificate authentication.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Certificate.xml:3]: Your version is fine, feel free to disregard my suggestion. You can do cert auth without this class, but that class makes it easier.

```suggestion
    <summary>Provides classes to support certificate authentication.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Cookies.xml [hunk 24]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Cookies">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for cookie based authentication.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Cookies.xml:3]:
```suggestion
    <summary>Contains types that support cookie based authentication.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Facebook.xml [hunk 25]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Authentication.Facebook">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types that enable support for facebook OAuth based authentication.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Facebook.xml:3]:
```suggestion
    <summary>Contains types that support facebook OAuth based authentication.</summary>
```

**Review by Rick-Anderson — APPROVED** [review]
Your version is fine. Don't really like better tho.

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Facebook.xml (hunk 25 above)
Comment (Rick-Anderson) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.Authentication.Facebook.xml:3]: If you agree with this suggestion, apply globally. Your version is more accurate, mine is more terse.

**Comment by HaoK** [comment]
I don't feel strongly about support/enables, so leaving things as is works for me if you are ok @Rick-Anderson

**Comment by JamesNK** [comment]
Thanks @HaoK!

**Pull request merged**